### ISSUE TITLE
Unexpected TypeError while running MC simulations

### ISSUE THREAD

**Issue by Mainv4** [glotzerlab/hoomd-blue#845]
# Description
I got unexpected TypeError as follow when I try to launch Monte Carol simulations: 
```python
~/.miniconda3/lib/python3.8/site-packages/hoomd/simulation.py in run(self, steps, write_at_start)
    284             raise RuntimeError('Cannot run before state is set.')
    285         if not self.operations._scheduled:
--> 286             self.operations._schedule()
    287
    288         self._cpp_sys.run(int(steps), write_at_start)

~/.miniconda3/lib/python3.8/site-packages/hoomd/operations.py in _schedule(self)
    194         sim = self._simulation
    195         if not (self.integrator is None or self.integrator._attached):
--> 196             self.integrator._attach()
    197         if not self.updaters._synced:
    198             self.updaters._sync(sim, sim._cpp_sys.updaters)

~/.miniconda3/lib/python3.8/site-packages/hoomd/hpmc/integrate.py in _attach(self)
    151                 self._simulation.device._cpp_msg.warning(
    152                     "Falling back on CPU. No GPU implementation for shape.\n")
--> 153             self._cpp_obj = getattr(_hpmc, self._cpp_cls)(sys_def, self.seed)
    154             self._cpp_cell = None
    155

TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. hoomd.hpmc._hpmc.IntegratorHPMCMonoSphere(arg0: hoomd._hoomd.SystemDefinition, arg1: int)

Invoked with: <hoomd._hoomd.SystemDefinition object at 0x7f911e616fb0>, 5314269779
```
And this error seems to appear in a random way. Indeed, by restarting the code, it can work. What can I do to avoid it in any case please ?

# Configuration
Platform:
Linux
CPU

Installation method:
Conda package

Versions:
Python version: 3.8
HOOMD-blue version: 3.0

**Comment by joaander** [comment]
The unit tests are passing for HPMC with spheres. Please post a minimal script that reproduces this error so we can determine what causes it.

**Comment by Mainv4** [comment]
Thank you for your response. Below is a script likely to reproduce the error. I was able to partially fix the problem by forcing the script to restart itself if an error occurred. Thanks again.

# Script
```python
def initSystem(c,L):
    V = L**3
    # Number of particles
    N_B = math.ceil(c * V * Na * 1e-27) 
    N_A = int(N_B/3.)
    N_particles = N_A + N_B
    # Size of the box
    K = math.ceil(N_particles**(1/3.))
    # Genereation of positions on a lattice
    x = numpy.linspace(-L / 2, L / 2, K, endpoint=False)
    position = list(itertools.product(x, repeat=3))
    numpy.random.shuffle(position)

    # drawing of the scene
    snapshot = gsd.hoomd.Snapshot()
    snapshot.particles.N = N_particles
    snapshot.particles.position = position[0:N_particles]
    snapshot.particles.typeid = [0]*int(N_particles/4.) + [1]*(N_particles-int(N_particles/4.))
    snapshot.configuration.box = [L, L, L, 0, 0, 0]
    snapshot.particles.types = ['A','B']
    snapshot.particles.diameter = [diameter_A]*int(N_particles/4.) + [diameter_B]*(N_particles-int(N_particles/4.))
    # writting of the scene
    with gsd.hoomd.open(name=path+'init.gsd', mode='wb') as f:
            f.append(snapshot)

def runSimulation(c,L):
    seed = numpy.random.randint(1e10)
    # Initialize Simualation
    cpu = hoomd.device.CPU()
    sim = hoomd.Simulation(device=cpu)
    sim.create_state_from_gsd(filename=path+'init.gsd')
    mc = hoomd.hpmc.integrate.Sphere(seed=seed,d=0.5)

    mc.shape['A'] = dict(diameter=diameter_A)
    mc.shape['B'] = dict(diameter=diameter_B)

    sim.operations.integrator = mc

    gsd = hoomd.write.GSD(filename=path+'trajectory.gsd',
                              trigger=hoomd.trigger.Periodic(100))
    sim.operations.writers.append(gsd)
    sim.run(100000, write_at_start = False)

initSystem(c,L)
runSimulation(c,L)

```

**Comment by joaander** [comment]
Thanks for providing the script. `pybind11` is throwing an error because it cannot convert a large seed to a 32-bit integer.  You need to choose seeds in the range [0, 2**32-1].

Ongoing work in #229 will change HOOMD seeding behavior so that seeds outside the supported range will result in a warning and not an error.

**Comment by joaander** [comment]
Fixed in #894. Seeds that are out of range are now truncated by taking the lowest valid bits and issuing a warning when seeds are provided outside the valid range.
